### PR TITLE
[UX] User Experience Improvements

### PR DESCRIPTION
**Pull request by reyna-abhyankar**
- [x] Earlier error handling for incorrect config or workflow paths
- [x] Rename `NoChange` to `Original` for `cognify evaluate`
- [x] Rename "pareto" to "optimization results" wherever applicable
- [x] Custom HotpotQA dataset loader, 5x faster than DSPy
- [x] Documentation for `llm_judge_factory` usage

### CODE REVIEW

File: cognify/optimizer/core/unified_layer_opt.py [hunk 1]
```diff
@@ -653,10 +653,10 @@ def post_optimize(self):
         pareto_frontier = self.get_pareto_front(candidates=candidates)
         if self.hierarchy_level == 0:
             print(f"================ Optimization Results =================") 
-            print(f"Num Pareto Frontier: {len(pareto_frontier)}")
+            print(f"# of Optimized Workflows Generated: {len(pareto_frontier)}")
```
Comment (yiying-zhang) [inline cognify/optimizer/core/unified_layer_opt.py:656]: better to still spell out Num

File: docs/source/user_guide/quickstart.rst [hunk 2]
```diff
@@ -253,29 +253,31 @@ An example output looks like this:
    (my_env) user@hostname:/path/to/quickstart$ cognify optimize workflow.py 
    > light_opt_layer | (best score: 0.16, lowest cost@1000: 0.09 $): 100%|███████████████| 10/10 [01:53<00:00, 11.30s/it]
    ================ Optimization Results =================
-   Num Pareto Frontier: 2
+   Number of Optimization Results: 2
    --------------------------------------------------------
-   Pareto_1
+   Optimization_1
+   Quality improves by 5%
```
Comment (yiying-zhang) [inline docs/source/user_guide/quickstart.rst:259]: better to use : like `Quality improvement: 5%`

File: docs/source/user_guide/quickstart.rst [hunk 3]
```diff
@@ -253,29 +253,31 @@ An example output looks like this:
...
+   Cost is 0.95x original
```
Comment (yiying-zhang) [inline docs/source/user_guide/quickstart.rst:260]: same as above

File: docs/source/user_guide/quickstart.rst [hunk 4]
```diff
@@ -253,29 +253,31 @@ An example output looks like this:
...
    Quality: 0.154, Cost per 1K invocation: $0.09
-   Applied at: light_opt_layer_6
    ========================================================
 
 The optimizer found two optimized workflow versions on the Pareto frontier, i.e., they are the most cost-effective solutions within all searched optimizations.
 
 .. note::
 
-   It's not guaranteed that the optimizer will find any better solutions than the original one. You might get ``Num Pareto Frontier: 0`` in the output.
+   It's not guaranteed that the optimizer will find any better solutions than the original one. You might get ``Number of Optimization Results: 0`` in the output.
```
Comment (yiying-zhang) [inline docs/source/user_guide/quickstart.rst:273]: shouldn't we be only display this message if the number of frontier is 0?

Comment (reyna-abhyankar) [reply docs/source/user_guide/quickstart.rst:273]: So should it not be in the documentation at all?

File: docs/source/user_guide/quickstart.rst (hunk 3 above)
Comment (reyna-abhyankar) [reply docs/source/user_guide/quickstart.rst:260]: Done

File: docs/source/user_guide/quickstart.rst (hunk 2 above)
Comment (reyna-abhyankar) [reply docs/source/user_guide/quickstart.rst]: Done

File: cognify/optimizer/core/unified_layer_opt.py (hunk 1 above)
Comment (reyna-abhyankar) [reply cognify/optimizer/core/unified_layer_opt.py]: Done

File: docs/source/user_guide/quickstart.rst (hunk 4 above)
Comment (yiying-zhang) [reply docs/source/user_guide/quickstart.rst:273]: oh.. i somehow didn't see that this is in documentation. then it's fine.

**Pull request merged**